modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2450,9 +2450,7 @@ def concat(
             joined_index = self.columns
             left_parts = self._partitions
             right_parts = [o._partitions for o in others]
-            new_lengths = self._row_lengths + [
-                length for o in others for length in o._row_lengths
-            ]
+            new_lengths = None
```

Comment:
Note that if we set `new_lengths` to None, then `new_lengths` will be computed in the constructor and will be the same as if we compute it via the removed line.

Reply:
Why do we need this change?

Reply:
Oh, I figured this out that is true only for other execution backends which we should support rebalancing for.

Reply:
Still, `new_lengths` will probably be computed in the constructor for pandas on ray as well. Is this correct?

Reply:
`new_lengths` should probably be computed in `rebalance_partitions`.

Reply:
It is a good point, and in fact `new_lengths` are computed in `rebalance_partitions` by assigning the length as metadata to the `virtual_partition` object. The `row_lengths` in this layer will pick up that value with no extra computation.

Reply:
As for `pandas on Ray`, this is okay because the `row_lengths` in this layer will pick up computed value in `rebalance_partitions` with no extra computation. However, this may slow down other execution backends, particularly pandas on Dask.

Reply:
Yes, that is a fair point. I think it will get resolved, but there may be a performance cost on further computations.